core/src/test/java/org/elasticsearch/index/query/MatchQueryBuilderTests.java

```diff
@@ -277,81 +273,68 @@ public void testBadAnalyzer() throws IOException {
                 "      \"fuzzy_transpositions\" : true,\n" +
                 "      \"lenient\" : false,\n" +
-                "      \"zero_terms_query\" : \"NONE\",\n" +
+                "      \"zero_terms_query\" : \"ALL\",\n" +
                 "      \"boost\" : 1.0\n" +
                 "    }\n" +
                 "  }\n" +
                 "}";
-        MatchQueryBuilder qb = (MatchQueryBuilder) parseQuery(json1);
-        checkGeneratedJson(expected, qb);
-
-        String json2 = "{\n" +
-                "    \"match\" : {\n" +
-                "        \"message\" : {\n" +
-                "            \"query\" : \"this is a test\",\n" +
-                "            \"type\" : \"phrase_prefix\"\n" +
-                "        }\n" +
-                "    }\n" +
-                "}";
-        qb = (MatchQueryBuilder) parseQuery(json2);
-        checkGeneratedJson(expected, qb);
-
-        String json3 = "{\n" +
-                "    \"match_phrase_prefix\" : {\n" +
-                "        \"message\" : {\n" +
-                "            \"query\" : \"this is a test\",\n" +
-                "            \"max_expansions\" : 10\n" +
-                "        }\n" +
-                "    }\n" +
-                "}";
-        expected = "{\n" +
+        MatchQueryBuilder qb = (MatchQueryBuilder) parseQuery(json);
+        checkGeneratedJson(json, qb);
+
+        assertEquals(json, "to be or not to be", qb.value());
+        assertEquals(json, Operator.AND, qb.operator());
+    }
+
+    public void testLegacyMatchPhrasePrefixQuery() throws IOException {
+        String json = "{\n" +
                 "  \"match\" : {\n" +
                 "    \"message\" : {\n" +
-                "      \"query\" : \"this is a test\",\n" +
+                "      \"query\" : \"to be or not to be\",\n" +
                 "      \"type\" : \"phrase_prefix\",\n" +
-                "      \"operator\" : \"OR\",\n" +
-                "      \"slop\" : 0,\n" +
+                "      \"operator\" : \"AND\",\n" +
+                "      \"slop\" : 2,\n" +
                 "      \"prefix_length\" : 0,\n" +
-                "      \"max_expansions\" : 10,\n" +
+                "      \"max_expansions\" : 30,\n" +
                 "      \"fuzzy_transpositions\" : true,\n" +
                 "      \"lenient\" : false,\n" +
-                "      \"zero_terms_query\" : \"NONE\",\n" +
+                "      \"zero_terms_query\" : \"ALL\",\n" +
                 "      \"boost\" : 1.0\n" +
                 "    }\n" +
                 "  }\n" +
                 "}";
-        qb = (MatchQueryBuilder) parseQuery(json3);
-        checkGeneratedJson(expected, qb);
+        MatchQueryBuilder qb = (MatchQueryBuilder) parseQuery(json, ParseFieldMatcher.EMPTY);
+        checkGeneratedJson(json, qb);
+
+        assertEquals(json, "to be or not to be", qb.value());
+        assertEquals(json, Operator.AND, qb.operator());
```

Comment:
an alternative would be parsing the proper version of the query, calling toQuery against both and checking out that they are the same via equals checks.

Also we could add here a quick serialization check using assertSerialization